### PR TITLE
Add Setup DVC Section

### PR DESCRIPTION
**Pull request by julieg18**
* moves setup dvc logic from experiments section to new dvc section

## Demo

https://user-images.githubusercontent.com/43496356/231002775-d9b0f08f-852e-43dd-acc2-610ececa1342.mov

Part of #3434

### CODE REVIEW

File: webview/src/setup/components/App.tsx [hunk 1]
```diff
@@ -84,20 +85,35 @@ export const App: React.FC = () => {
...
+      </SetupContainer>
+      <SetupContainer
+        sectionKey={SetupSection.EXPERIMENTS}
+        title="Experiments"
+        sectionCollapsed={sectionCollapsed}
+        setSectionCollapsed={setSectionCollapsed}
+      >
+        <Experiments
+          isDvcSetup={
+            projectInitialized && Boolean(cliCompatible) && !needsGitCommit
```
Comment (julieg18) [inline webview/src/setup/components/App.tsx:113]: Is there a better way to check for if DVC is setup properly?

File: extension/package.json [hunk 2]
```diff
@@ -1449,7 +1449,7 @@
       },
       {
         "view": "dvc.views.welcome",
-        "contents": "New to the extension?\n[Show Walkthrough](command:dvc.getStarted)\n\nThe extension is currently unable to initialize.\n[Show Setup](command:dvc.showExperimentsSetup)",
+        "contents": "New to the extension?\n[Show Walkthrough](command:dvc.getStarted)\n\nThe extension is currently unable to initialize.\n[Show Setup](command:dvc.showDvcSetup)",
```
Comment (mattseddon) [inline extension/package.json:1452]: [Q] What do the "Show Experiments", "Show Plots" and "Run Experiment" buttons redirect to when the CLI is not setup? I think it should be the new section.

See `showSetupOrExecuteCommand`.

File: webview/src/setup/components/Experiments.tsx [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
+  setSectionCollapsed
 }) => {
+  if (!isDvcSetup) {
+    return (
+      <EmptyState isFullScreen={false}>
+        <h1>DVC is not setup</h1>
+        <p>DVC needs to be setup before you can access experiments.</p>
+        <Button
+          onClick={() =>
+            setSectionCollapsed({
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply webview/src/setup/components/Experiments.tsx:28]: Could hide or "grey out" the sections that are currently unavailable.

**Review by mattseddon — APPROVED** [review]
We need to fix the redirects before merging. 

The separation between the sections needs to be well-defined. After setup is complete I think we want to show the details of where the CLI has been located much like the tooltip that we show for the status bar item: https://github.com/iterative/vscode-dvc/blob/21ef1dd990a2751d29b2074001b520b674c509ef/extension/src/status.ts#L107

Seeing as we have a lot more room than a tooltip and we can make things far more obvious we want to show the version of the CLI/Min required version etc as this is very useful information that we can use to get people "unstuck". We can also go in the direction of removing the toast messages and moving them into the setup screen. Toast messages are here: https://github.com/iterative/vscode-dvc/blob/21ef1dd990a2751d29b2074001b520b674c509ef/extension/src/cli/dvc/discovery.ts#L1

File: webview/src/setup/components/App.tsx [hunk 4]
```diff
@@ -84,20 +85,35 @@ export const App: React.FC = () => {
...
+        <Dvc
           canGitInitialize={canGitInitialize}
           cliCompatible={cliCompatible}
-          hasData={hasData}
           isPythonExtensionInstalled={isPythonExtensionInstalled}
           needsGitInitialized={needsGitInitialized}
           needsGitCommit={needsGitCommit}
           projectInitialized={projectInitialized}
           pythonBinPath={pythonBinPath}
+          hasData={hasData}
```
[the remark this answers is not in the document]
Comment (julieg18) [reply webview/src/setup/components/App.tsx:101]: > [Q] What does hasData have to do with setting up the CLI? Isn't this for experiments only? 

`hasData` is used for when the `DVC` section knows what to do with the `Show Experiments` button that appears when Setup is complete. Though we could rename the prop to make things more clear! 

> Same question for needsGitCommit?

Makes sense! Will move the `needsGitCommit` logic to `Experiments`.

File: webview/src/setup/components/Dvc.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          hasData
+            ? showExperiments
+            : () =>
+                setSectionCollapsed({
+                  dvc: true,
+                  experiments: false,
+                  studio: true
+                })
+        }
+        text="Show Experiments"
```
[the remark this answers is not in the document]
Comment (julieg18) [reply webview/src/setup/components/Dvc.tsx:84]: Good idea! Though not sure how to implement it. I thought I could do this with an effect hook, but that breaks `SETUP_SHOW_STUDIO` and `SETUP_SHOW_EXPERIMENTS`. Will put some more thought into this one and do it in a followup!

**Comment by julieg18** [comment]
Thanks for the reviews, everyone! 

What's Left To Do:

- [ ] Hide or "grey out" the sections that are currently unavailable. (https://github.com/iterative/vscode-dvc/pull/3655#discussion_r1162154764)
- [ ] Have a section automatically close and open another when setup finished (https://github.com/iterative/vscode-dvc/pull/3655#discussion_r1162154053)
- [ ] Have DVC section show more details about the dvc cli version (number, location, etc.) in the "Setup Complete" screen (https://github.com/iterative/vscode-dvc/pull/3655#pullrequestreview-1378227661)
- [ ] Move some of dvc setup messaging that is shown in toasts to the DVC section (https://github.com/iterative/vscode-dvc/pull/3655#pullrequestreview-1378227661)

All of these items are going to be in separate prs since they are going to require extra reviews. I'm currently working on the first one, but I think we're good to merge! Will wait a bit to make sure there are no objections :)

**Pull request merged**